frontend/test/e2e/ContractExampleStorage.spec.ts

```diff
@@ -1,26 +1,46 @@
 import { WebDriver, By, until } from 'selenium-webdriver'
 import { ContractsPage } from '../pages/ContractsPage.js'
 import { RunDebugPage } from '../pages/RunDebugPage.js'
+import { SettingsPage } from '../pages/SettingsPage.js'
 import { before, describe, after, it } from 'node:test'
 import { expect } from 'chai'
-import { getDriver } from '../utils/driver.js'
+import { PageFactory } from '../utils/pages.js'
 
 let driver: WebDriver
 let contractsPage: ContractsPage
 let runDebugPage: RunDebugPage
+let settingsPage: SettingsPage
 
 describe('Contract Example Storage', () => {
   before(async () => {
-    driver = await getDriver()
-    await driver.manage().setTimeouts({ implicit: 10000 })
-    contractsPage = new ContractsPage(driver)
-    runDebugPage = new RunDebugPage(driver)
+    driver = await PageFactory.getDriver()
+    contractsPage = await PageFactory.getContractsPage()
+    runDebugPage = await PageFactory.getRunDebugPage()
+    settingsPage = await PageFactory.getSettingsPage()
+
+    await contractsPage.navigate()
+    await contractsPage.waitUntilVisible()
+    await contractsPage.skipTutorial()
+    await settingsPage.navigate()
+
+    const initialValidators = await settingsPage.getValidatorsElements()
+    if (initialValidators.length < 1) {
+      await settingsPage.createValidator({
+        provider: 'heuristai',
+        model: 'mistralai/mixtral-8x7b-instruct',
+        stake: 7
+      })
+      const existingValidators = await settingsPage.getValidatorsElements()
+      expect(
+        existingValidators.length,
+        'number of validators should be greather than old validators list'
+      ).be.greaterThan(initialValidators.length)
+    }
```

Comment:
@bradleySuira my point was to make this piece of code reusable on all tests. I was referring to the validators' creation and initialization

Reply:
got it all fixed now and created a method inside the `SettingsPage`